code/linalg.c

```diff
@@ -274,16 +274,16 @@ mp_obj_t linalg_eye(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args)
     ndarray_obj_t *ndarray = create_new_ndarray(m, n, dtype);
     mp_obj_t one = mp_obj_new_int(1);
     size_t i = 0;
-    if((k >= 0) && (k < n)) {
-        while(k < n) {
+    if((k >= 0) && (abs(k) < n)) {
+        while(abs(k) < n) {
             mp_binary_set_val_array(dtype, ndarray->array->items, i*n+k, one);
             k++;
             i++;
         }
-    } else if((k < 0) && (-k < m)) {
+    } else if((k < 0) && (abs(k) < m)) {
         k = -k;
         i = 0;
-        while(k < m) {
+        while(abs(k) < m) {
```

Comment:
Is the absolute value necessary here?

Reply:
nope, I'll take it out

Reply:
OK. (Might be a bit expensive with the abs function, but I don't think it is a deal-breaker.)